lib/replica.rb

```diff
@@ -150,29 +150,30 @@ def do_query(endpoint, query)
     Net::HTTP::get(URI.parse(url))
   end
 
+  # Query URL for the WikiEduDashboardTools repository
   def compile_query_url(endpoint, query)
     base_url = 'http://tools.wmflabs.org/wikiedudashboard/'
-    "#{base_url}#{endpoint}?#{language_and_project_params}&#{query}"
+    "#{base_url}#{endpoint}?lang=#{@wiki.language}&project=#{@wiki.project}\
+    &db=#{db_params}&#{query}"
   end
 
-  # We must special-case Multilingual Wikisource/ Wikimedia Incubator because
-  # their labs database names do not follow the normal pattern; they are called
-  # 'sourceswiki' and 'incubatorwiki' respectively, which are the same as we'd
-  # get if a Wikipedia's language was "sources" or "incubator".
-  def language_and_project_params
-    if @wiki.project == 'wikisource' && @wiki.language.nil?
-      'lang=sources&project=wikipedia'
-    elsif @wiki.project == 'wikimedia' && @wiki.language == 'incubator'
-      "lang=incubator&project=wikipedia"
-    else
-      "lang=#{@wiki.language}&project=#{@wiki.project}"
-    end
-  end
 
   def compile_usernames_query(users)
     { usernames: users.map(&:username) }.to_query
   end
 
+  # Returns special Labs database names for databases not meeting
+  # project/language naming conventions, else nil
+  def db_params
+    if @wiki.project == 'wikidata'
+      'wikidatawiki'
```

Comment:
Instead of just doing the db value, you could return the whole query param, `db=wikidatawiki` and such, and then fall back to `"lang=#{@wiki.language}&project=#{@wiki.project}"` at the end.